modules/selenium/src/main/java/org/testcontainers/containers/BrowserWebDriverContainer.java

```diff
@@ -212,9 +211,8 @@ protected void finished(Description description) {
         this.stop();
     }
 
-    private void stopAndRetainRecording(Description description) {
-        File recordingFile = new File(vncRecordingDirectory, "recording-" + filenameDateFormat.format(new Date()) + ".flv");
-
+    private void stopAndRetainRecording(Description description, boolean succeeded) {
+        File recordingFile = recordingFileFactory.recordingFileForTest(vncRecordingDirectory, description, succeeded);
         LOGGER.info("Screen recordings for test {} will be stored at: {}", description.getDisplayName(), recordingFile);
```

Comment:
I know this is not part of your PR, but this line will fail for none JUnit usages (since no `Description` will be available). Is the `Description` actually needed?

Reply:
Ok. Is there a test-case for that in the codebase already?
Or better, if the description is null what should the factory default to?

Reply:
I see you are using `Description` for the values of your recorder as well. I think it's okay for now, since we won't break the current API (which is kind of JUnit4 focused after all). We can refactor this part in the future. I'll check if it works like this with testcontainers-spock out of the box.